src/combat.ts

```diff
@@ -130,52 +145,262 @@ export class Macro extends LibramMacro {
+        .attack()
+        .repeat()
+    );
+  }
+
+  static kill(): Macro {
+    return new Macro().kill();
+  }
+
+  basicCombat(): Macro {
+    return this.startCombat().kill();
+  }
+
+  static basicCombat(): Macro {
+    return new Macro().basicCombat();
+  }
+
+  ghostBustin(): Macro {
+    // Only bust ghosts if you have enough stunners to prevent getting hit
+    let stunRounds = 0;
+    let classStun: Skill | null = null;
+    let extraStun: Skill | null = null;
+    if (have($item`Rain-Doh blue balls`)) stunRounds++;
+    if (get("lovebugsUnlocked")) stunRounds++;
+    if (
+      myClass() === $class`Seal Clubber` &&
+      have($skill`Club Foot`) &&
+      myMp() >= mpCost($skill`Club Foot`)
+    ) {
+      const clubRounds =
+        Math.min(myFury(), 3) + (itemType(equippedItem($slot`weapon`)) === "club" ? 1 : 0) - 1;
+      if (stunRounds > 0) {
+        classStun = $skill`Club Foot`;
+        stunRounds += clubRounds;
+      }
+    } else if (
+      myClass() === $class`Turtle Tamer` &&
+      have($skill`Shell Up`) &&
+      myMp() >= mpCost($skill`Shell Up`)
+    ) {
+      const shellRounds =
+        (have($effect`Blessing of the Storm Tortoise`) ? 2 : 0) +
+        (have($effect`Grand Blessing of the Storm Tortoise`) ? 3 : 0) +
+        (have($effect`Glorious Blessing of the Storm Tortoise`) ? 4 : 0);
+      if (shellRounds > 0) {
+        classStun = $skill`Shell Up`;
+        stunRounds += shellRounds;
+      }
+    } else if (
+      myClass() === $class`Pastamancer` &&
+      have($skill`Entangling Noodles`) &&
+      myMp() >= mpCost($skill`Entangling Noodles`)
+    ) {
+      classStun = $skill`Entangling Noodles`;
+      stunRounds += 2;
+    } else if (myClass() === $class`Sauceror` && have($skill`Soul Bubble`) && mySoulsauce() >= 5) {
+      classStun = $skill`Soul Bubble`;
+      stunRounds += 2;
+    } else if (
+      myClass() === $class`Accordion Thief` &&
+      have($skill`Accordion Bash`) &&
+      itemType(equippedItem($slot`weapon`)) === "accordion" &&
+      myMp() >= mpCost($skill`Accordion Bash`)
+    ) {
+      classStun = $skill`Accordion Bash`;
+      stunRounds += 2;
+    } else if (myClass() === $class`Disco Bandit`) {
+      // Rave Knockout seems like a pain
+    }
+
+    // Don't use shadow noodles unless we really need it.
+    if (
+      stunRounds < 3 &&
+      classStun !== $skill`Entangling Noodles` &&
+      have($skill`Shadow Noodles`) &&
+      myMp() >= mpCost(classStun ?? $skill`none`) + mpCost($skill`Shadow Noodles`)
+    ) {
+      extraStun = $skill`Shadow Noodles`;
+      stunRounds += 2;
+    }
+
+    // Lacking multi-round stuns
+    if (stunRounds < 3) {
+      return this.basicCombat();
+    }
+
+    return this.tryHaveSkill($skill`Sing Along`)
+      .tryHaveItem($item`Rain-Doh blue balls`)
+      .externalIf(get("lovebugsUnlocked"), Macro.trySkill($skill`Summon Love Gnats`))
+      .tryHaveSkill(classStun)
```

Comment:
Does this  behave nicely when classStun is null?

Reply:
yes, I changed tryHaveSkill to accept null